File: Lecture 3/app/src/main/java/com/example/bmi/Score.java
package com.example.bmi;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

public class Score extends AppCompatActivity {
    TextView result;
    String answer = " Your score is: " + Quiz.getScore();
    Button home_button, stats_button;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_score);
        result = findViewById(R.id.result);
        result.setText(answer);

        home_button = (Button) findViewById(R.id.go_to_home);
        home_button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v){
                go_to_home();
            };
         });

        stats_button = (Button) findViewById(R.id.go_to_statistics);
        stats_button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v){
                go_to_stats();
            };
        });

    }

    public void go_to_home(){
        Intent intent = new Intent(this, MainActivity.class);
        startActivity(intent);
    };

    public void go_to_stats(){
        Intent intent = new Intent(this, Stats.class);
        startActivity(intent);
    };
}

File: Lecture 4/bmi/app/src/main/java/com/example/bmi/Score.kt
package com.example.bmi

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class Score : AppCompatActivity() {
    var result: TextView? = null
    var answer = " Your score is: " + Quiz.getScore()
    var home_button: Button? = null
    var stats_button: Button? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_score)
        result = findViewById(R.id.result)
        result.setText(answer)
        home_button = findViewById<View>(R.id.go_to_home) as Button?
        home_button.setOnClickListener(object : OnClickListener() {
            fun onClick(v: View?) {
                go_to_home()
            }
        })
        stats_button = findViewById<View>(R.id.go_to_statistics) as Button?
        stats_button.setOnClickListener(object : OnClickListener() {
            fun onClick(v: View?) {
                go_to_stats()
            }
        })
    }

    fun go_to_home() {
        val intent = Intent(this, MainActivity::class.java)
        startActivity(intent)
    }

    fun go_to_stats() {
        val intent = Intent(this, Stats::class.java)
        startActivity(intent)
    }
}